tests/Unit/Updates/UpdateInterruptionGuardTest.php

```
@@ -0,0 +1,409 @@
+    {
+        Log::spy();
+
+        $manager = new RecordingApplicationUpdateManager;
+        $manager->simulateInFlight( UpdateStep::Migrations );
+
+        $manager->handleInterruptedUpdate();
+
+        $this->assertContains( 'disable', $manager->calls );
+        $this->assertFalse( $manager->believesMaintenanceModeIsActive() );
+
+        Log::shouldHaveReceived( 'critical' );
```

PHPStan (2.2.6)

[error] Call to an undefined static method Illuminate\Support\Facades\Log::shouldHaveReceived().


(staticMethod.notFound)